src/modules/idyntree-visualizer/Visualizer.cpp

```diff
@@ -0,0 +1,225 @@
+#include "Visualizer.h"
+
+using namespace std::chrono_literals;
+
+bool idyntree_yarp_tools::Visualizer::configure()
+{
+    // Listen to signals for closing in a clean way the application
+    idyntree_yarp_tools::handleSignals([this](){this->closeSignalHandler();});
+
+    std::lock_guard<std::mutex> lock(m_mutex);
+    std::string pathToModel = yarp::os::ResourceFinder::getResourceFinderSingleton().findFileByName("model.urdf");
+    modelLoader.loadReducedModelFromFile(pathToModel, jointList);
+
+    viz.init(options);
+    textureInterface = viz.textures().add("AdditionalTexture", textureOptions);
+
+    viz.camera().setPosition(iDynTree::Position(1.2, 0.0, 0.5));
+    viz.camera().setTarget(iDynTree::Position(-0.15, 0.0, 0.15));
+    viz.camera().animator()->enableMouseControl(true);
+
+    double sqrt2 = std::sqrt(2.0);
+    viz.enviroment().lightViz("sun").setDirection(iDynTree::Direction(0.5/sqrt2, 0, -0.5/sqrt2));
+    viz.enviroment().addLight("secondSun");
+    viz.enviroment().lightViz("secondSun").setType(iDynTree::LightType::DIRECTIONAL_LIGHT);
+    viz.enviroment().lightViz("secondSun").setDirection(iDynTree::Direction(-0.5/sqrt2, 0, -0.5/sqrt2));
+
+    textureInterface->environment().lightViz("sun").setDirection(iDynTree::Direction(-0.5/sqrt2, 0, -0.5/sqrt2));
+    textureInterface->environment().addLight("secondSun");
+    textureInterface->environment().lightViz("secondSun").setType(iDynTree::LightType::DIRECTIONAL_LIGHT);
+    textureInterface->environment().lightViz("secondSun").setDirection(iDynTree::Direction(-0.5/sqrt2, 0, -0.5/sqrt2));
+    textureInterface->environment().setElementVisibility("floor_grid", false);
+    textureInterface->environment().setElementVisibility("world_frame", false);
+    textureInterface->environment().setBackgroundColor(iDynTree::ColorViz(0.0, 0.0, 0.0, 0.0));
+
+    viz.addModel(modelLoader.model(), "robot");
+
+    image.setPixelCode(VOCAB_PIXEL_RGB);
+    image.resize(textureOptions.winWidth, textureOptions.winHeight);
+
+    // initialise yarp network
+    yarp::os::Network yarpNetwork;
+    if (useNetwork && !yarpNetwork.checkNetwork())
+    {
+        yWarning()<<"No YARP network found. Avoiding to use the network.";
+        useNetwork = false;
+        connectToRobot = false;
```

Comment:
I guess this is like that because at the moment `useNetwork` is not configurable, right?

Reply:
Actually, ``useNetwork`` is set to ``true`` now. First it checks if the network is available. If not, it is set to ``false`` automatically.

Reply:
Yes, but in my experience this kind of logic is quite fragile. The time that yarpserver for some reason is down and you actually want it up, this logic delays the moment that you realize that it is down, however I understand it needs to be like that if `useNetwork` is not configurable from the user.

Reply:
It will be configurable. For the moment I hardcoded everything, but the idea is to get every parameter from configuration file. I was a bit in doubt if it was worth between a tank-like behavior, where, no matter what, the robot is visualized, or using a fail-fast approach, where if something is not as supposed to be, it breaks.

In the end, I opted for the tank solution. I was imagining the case where the configuration file of the visualizer is going to be modified pretty seldom, and the user may want the visualizer to run without many complaints.

What would be your suggestion?

Reply:
I personally prefer to fail fast, as in my experience the time spent in debugging when something is not working as expected is higher then the time spent modifying a conf file or a command line option to specify exactly how do you want a tool to work, but this is not a big deal.

Reply:
Ok, no big deal, I edit it.

Reply:
Used fail-fast in https://github.com/robotology/idyntree-yarp-tools/pull/4/commits/d4f88245f08abd14522a9f7030d1776233cba896

Let me know if you are ok with merging.